artemis-commons/src/main/java/org/apache/activemq/artemis/utils/ArtemisSystemProperties.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.activemq.artemis.utils;
+
+import java.net.URL;
+import java.util.Properties;
+
+import org.jboss.logging.Logger;
+
+public final class ArtemisSystemProperties {
+
```

Comment:
This class is a bad idea. Having config setting to statics means if two cfs and want different setting for each eg connection to different clusters, then you cant.

Reply:
Also this will act as a broken window and encourage further set this way, further causing such issue.

Reply:
This class is about defaults... The default coming from a system property.

There's no way around a System property. users coded for older servers will need to be switched without changes.


The idea of this class is to use a file on the classLoader.

This is mainly for wildfly. In Wildfly they have set the entire VM to use older properties. ActiveMQJMSClient should be deprecated... as of now I don't see a way we could fix this without System properties.

Reply:
BTW: Two CFs can still have different settings.. on the CF this is just altering the default. You can still switch independently on the CF. The default now comes from a system property instead of a constant variable.


As of on the ActiveMQJMSClient, there's no way to alter the semantic without a system property. But users shouldn't be encouraged to use that class anyways. I'm considering to set it as deprecated.

Reply:
>>> Also this will act as a broken window and encourage further set this way, further causing such issue.


The system property should change defaults only. Only exception will be the AtiveMQJMSClient.

I don't see any difference than getting a System.property on the class itself instead of encapsulating usage here.

Reply:
My concern is the broken window here. That further people may start abusing this in future developments. Could this be closer coupled with the defaults class or something so its very clear this is only used to override the default

Reply:
The reason I encapsulated this.. was because of the file load on the classPath. Wildfly (or anyone else) will be able to specify a properties file on the classLoader and alter this as a System property.

I am also reusing that between the default on CF and the ActiveMQJMSClient (that needs a system property regardless).

Reply:
This is for JMS users only could all the 1x changes just sit in the jms module, rather thsn spreading out